integration-tests/tests/src/tests/sync/flexible.ts

```diff
@@ -437,6 +442,264 @@ describe("Flexible sync", function () {
+            callback,
+          );
+
+          expect(callback.callCount).equals(1);
+
+          expect(callback.withArgs(1.0).calledOnce).to.be.true;
+        });
+
+        it("should be called multiple times with different values during uploads", async function (this: RealmContext) {
+          const realm = this.realm;
+          // eslint-disable-next-line @typescript-eslint/no-empty-function, @typescript-eslint/no-unused-vars
+          const callback = spy((estimate: number) => {});
+
+          realm.syncSession?.addProgressNotification(
+            Realm.ProgressDirection.Upload,
+            Realm.ProgressMode.ReportIndefinitely,
+            callback,
+          );
+
+          realm.write(() => {
+            for (let i = 0; i < 3; i++) {
+              realm.create(Person, {
+                _id: new BSON.ObjectId(),
+                name: "Person",
+                age: i,
+              });
+            }
+          });
+
+          await realm.syncSession?.uploadAllLocalChanges();
+
+          // There should be at least one point where the progress is not yet finished.
+          expect(callback.args.find(([estimate]) => estimate < 1)).to.not.be.undefined;
+
+          expect(callback.withArgs(1.0).called).to.be.true;
+        });
+      });
+
+      describe(`with ProgressDirection.Download`, function () {
+        it("should only call the callback once with 1.0 when there is nothing to download", async function (this: RealmContext) {
+          const realm = this.realm;
+          // eslint-disable-next-line @typescript-eslint/no-empty-function, @typescript-eslint/no-unused-vars
+          const callback = spy((estimate: number) => {});
+
+          realm.syncSession?.addProgressNotification(
+            Realm.ProgressDirection.Download,
+            ProgressMode.ReportIndefinitely,
+            callback,
+          );
+
+          expect(callback.callCount).equals(1);
+
+          expect(callback.withArgs(1.0).calledOnce).to.be.true;
+        });
+
+        it("should be called multiple times with different values during downloads", async function (this: RealmContext &
+          AppContext) {
+          const realm = this.realm;
+          // eslint-disable-next-line @typescript-eslint/no-empty-function, @typescript-eslint/no-unused-vars
+          const callback = spy((estimate: number) => {});
+          const realm2 = await Realm.open({
+            schema: [Person, Dog],
+            sync: {
+              flexible: true,
+              user: await this.app.logIn(Credentials.anonymous()),
+              newRealmFileBehavior: {
+                type: OpenRealmBehaviorType.OpenImmediately,
+              },
+            },
+          });
+
+          realm2.syncSession?.addProgressNotification(
+            Realm.ProgressDirection.Download,
+            Realm.ProgressMode.ReportIndefinitely,
+            callback,
+          );
+
+          await realm2.objects(Person).subscribe();
```

Comment:
can't think of a good way to have a proper download-upload situation, any ideas? another weirdness is that the estimates are
[ 1, 1, 1, 1, 1, 1, ... 1, ] no real progress
uploads on the other hand are more reasonable [1, 0, 0.3123, 0,4112.... ,1]